proposed/2023/allow-version-property-in-projectreference.md

```diff
@@ -0,0 +1,107 @@
+# Allow users to define version ranges for ProjectReferences
+
+- [Martin Ruiz](https://github.com/martinrrm)
+- Start Date (2023-01-01)
+- [5556](https://github.com/NuGet/Home/issues/5556)
+
+# Summary
+
+Add a `Version` to `ProjectReference` tag in CSPROJ, to allow customers to specify the referenced project version in the `.nupkg` and `nuspec` files when doing a pack command.
+
+# Motivation
+
+When using `ProjectReference` there is no option to define a Version to the reference like in `PackageReference` and the version will always be defined as `>= ProjectVersion`, this results in customers manually modifying the nuspec file if they want to declare a different version than the project version.
+
+# Explanation
+
+Currently when adding a `ProjectReference` to a project, there is no property to specify which version(s) of it to be used when doing a pack. 
+When doing a pack to a package with a `ProjectRefernce` it will always be added as a range, where the minumum version will be the ProjectReference version and with an open maximum version.
+
+```
+<ItemGroup>
+    <ProjectReference Include="../MyReferencedPackage/MyReferencedPackage.csproj" />
+</ItemGroup>
+```
+
+Add a `Version` property to `ProjectReference` and store that value in the assets file when doing a restore so we can retrieve that information when doing a `pack command.
+
+If there is no `Version` information then the behavior should be the current one.
+
+## Example
+
+### .CSPROJ file
+```
+<ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="[9.0.0, 13.0.2)" />
+    <ProjectReference Include="../MyReferencedPackage/MyReferencedPackage.csproj" Version="[1.0.0, 2.0.0)" />
+</ItemGroup>
+```
+
+### assets file
+```
+"frameworks": {
+    "net6.0": {
+        "targetAlias": "net6.0",
+        "projectReferences": {
+            "C:\\Users\\mruizmares\\source\\repos\\ConsoleApp1\\MyReferencedPackage\\MyReferencedPackage.csproj": {
+                "projectPath": "C:\\Users\\mruizmares\\source\\repos\\ConsoleApp1\\MyReferencedPackage\\MyReferencedPackage.csproj",
+                "version": "[1.0.0, 2.0.0)"
```

Comment:
Will this change be compatible with old version tools?
That is, will old tool be able to build the solution/project with the new format assets file?
You may refer to https://github.com/NuGet/NuGet.Client/pull/4654#discussion_r891833840 for similar checking.

Reply:
Tested doing a `dotnet build` and it works, you know if there are more ways to test if it is compatible?

Reply:
Did you use `--no-restore`? Otherwise `dotnet build` will auto-restore and overwrite your manually edited assets file.

Reply:
My guess is that because it's additive it should work.